Source/WebKit/UIProcess/Automation/WebDriverBidiProcessor.cpp

```diff
@@ -111,19 +117,183 @@ void WebDriverBidiProcessor::sendMessageToFrontend(const String& message)
 }
 
 
-// MARK: Inspector::BrowsingContextDispatcherHandler methods.
+// MARK: Inspector::BidiBrowsingContextDispatcherHandler methods.
 
-void WebDriverBidiProcessor::navigate(const BrowsingContext& browsingContext, const String& url, std::optional<Inspector::Protocol::BidiBrowsingContext::ReadinessState>&&, CommandCallbackOf<String, String>&& callback)
+void WebDriverBidiProcessor::activate(const BrowsingContext& browsingContext, CommandCallback<void>&& callback)
 {
     RefPtr session = m_session.get();
     if (!session)
         ASYNC_FAIL_WITH_PREDEFINED_ERROR(InternalError);
 
-    ASYNC_FAIL_WITH_PREDEFINED_ERROR(NotImplemented);
+    RefPtr webPageProxy = session->webPageProxyForHandle(browsingContext);
+    if (!webPageProxy)
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR(WindowNotFound);
```

Comment:
i wonder if it'd be helpful to have a `ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF` so that this can just be written as
```cpp
ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!webPageProxy, WindowNotFound);
```
personally i think that'd be cleaner